com.github.camel-tooling.lsp.eclipse.client.tests.ui/tests/com.github.cameltooling.lsp.ui.tests/src/com/github/cameltooling/lsp/ui/tests/CamelCatalogVersionFeatureTest.java

```diff
@@ -0,0 +1,183 @@
+import org.eclipse.reddeer.requirements.cleanerrorlog.CleanErrorLogRequirement;
+import org.eclipse.reddeer.swt.api.Shell;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem;
+import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
+import org.eclipse.reddeer.workbench.handler.WorkbenchShellHandler;
+import org.eclipse.reddeer.workbench.impl.editor.DefaultEditor;
+import org.eclipse.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import com.github.cameltooling.lsp.reddeer.editor.SourceEditor;
+import com.github.cameltooling.lsp.reddeer.preference.CamelCatalogVersion;
+
+/*
+*
+* @author fpospisi
+*/
+@RunWith(RedDeerSuite.class)
+public class CamelCatalogVersionFeatureTest {
+
+	public static final String PROJECT_NAME = "catalog-feature-test";
+	public static final String CAMEL_CONTEXT = "camel-context.xml";
+	public static final String RESOURCES_CONTEXT_PATH = "resources/catalog-version-feature-context.xml";
+	public static final String COMPONENT = "file-watch";
+	public static final String COMPONENT_PROPOSAL = "file-watch:path";
+	public static final String OLDER_VERSION = "2.15.1";
+
+	public static final String SOURCE_TAB = "Source";
+
+	private SourceEditor sourceEditor;
+	private int cursorPosition;
+
+	private String timePeriodfactor;
+	private static final String TIMEOUT_PERIOD_FACTOR_PROPETY_NAME = RedDeerProperties.TIME_PERIOD_FACTOR.getName();
+
+	/**
+	 * Creates empty project, then creates XML file with camel-context.
+	 */
+	@BeforeClass
+	public static void setupTestEnvironment() {
+		JavaProjectFactory.create(PROJECT_NAME);
+		new ProjectExplorer().selectProjects(PROJECT_NAME);
+		CreateNewEmptyFile.XMLFile(CAMEL_CONTEXT);
+		DefaultEditor editor = new DefaultEditor(CAMEL_CONTEXT);
+		editor.activate();
+		new DefaultCTabItem(SOURCE_TAB).activate();
+		EditorManipulator.copyFileContentToXMLEditor(RESOURCES_CONTEXT_PATH);
+	}
+
+	/**
+	 * Prepares test environment.
+	 */
+	@Before
+	public void defaultBeforeTestSetup() {
+		new CleanErrorLogRequirement().fulfill();
+	}
+
+	@Before
+	public void setupTimeout() {
+		timePeriodfactor = System.getProperty(TIMEOUT_PERIOD_FACTOR_PROPETY_NAME);
+		System.setProperty(TIMEOUT_PERIOD_FACTOR_PROPETY_NAME, "3");
+		TimePeriod.updateFactor();
+	}
+
+	/**
+	 * Cleans up test environment.
+	 */
+	@After
+	public void defaultAfterTestClean() {
+		WorkbenchShellHandler.getInstance().closeAllNonWorbenchShells();
+	}
+
+	@After
+	public void tearDown() {
+		if (timePeriodfactor != null) {
+			System.setProperty(TIMEOUT_PERIOD_FACTOR_PROPETY_NAME, timePeriodfactor);
+		} else {
+			System.clearProperty(TIMEOUT_PERIOD_FACTOR_PROPETY_NAME);
+		}
+		TimePeriod.updateFactor();
+	}
```

Comment:
Shouldn't we set back the camel catalog version to default value or value before starting test to avoid potential interaction with otrher tests?

Reply:
Good point Aurelien! I missed that, thank you.

@pospisilf Could you please update test to reverse camel catalog version? thank you!